dynamiqs/result.py

```diff
@@ -72,16 +72,15 @@ def extra(self) -> PyTree | None:
 
     def __str__(self) -> str:
         parts = {
-            'Solver  ': type(self.solver).__name__,
-            'Gradient': (
-                type(self.gradient).__name__ if self.gradient is not None else None
-            ),
+            'Solver  ': self.solver,
+            'Gradient': (self.gradient if self.gradient is not None else None),
             'States  ': array_str(self.states),
             'Expects ': array_str(self.expects) if self.expects is not None else None,
             'Extra   ': (
                 eqx.tree_pformat(self.extra) if self.extra is not None else None
             ),
             'Infos   ': self.infos if self.infos is not None else None,
+            'Options ': self.options,
         }
```

Comment:
It does not really look good 😢
```python
import dynamiqs as dq
import jax.numpy as jnp

n = 16
a = dq.destroy(n)
ad = dq.dag(a)

H = 10 * ad @ a
jump_ops = [jnp.sqrt(0.1) * a]
psi0 = dq.coherent(n, 1.0)
tsave = jnp.linspace(0, 1.0, 101)
gradient = dq.gradient.Autograd()
result = dq.mesolve(H, jump_ops, psi0, tsave, gradient=gradient)
print(result)
```

```
==== Result ====
Solver  : Tsit5(rtol=f32[], atol=f32[], safety_factor=f32[], min_factor=f32[], max_factor=f32[], max_steps=i32[])
Gradient: Autograd()
States  : Array complex64 (101, 16, 16) | 202.00 Kb
Infos   : 67 steps (67 accepted, 0 rejected)
Options : Options(save_states=bool[], verbose=bool[], cartesian_batching=bool[], t0=None, save_extra=None)
```

I think the reason is that for the JIT every int/bool is converted to a single-element JAX array 😬

For now I propose to keep the old result code, and we can think about a fix in another PR.